app/src/main/kotlin/zeta/android/apps/ui/activity/NavigationActivity.kt

```diff
@@ -0,0 +1,190 @@
+package zeta.android.apps.ui.activity
+
+import android.app.ActivityManager
+import android.os.Bundle
+import android.support.design.widget.AppBarLayout
+import android.support.design.widget.NavigationView
+import android.support.v4.content.ContextCompat
+import android.support.v4.view.GravityCompat
+import android.support.v4.widget.DrawerLayout
+import android.support.v7.app.ActionBarDrawerToggle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.Toolbar
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+
+import javax.inject.Inject
+
+import butterknife.BindView
+import dagger.android.AndroidInjection
+import zeta.android.apps.R
+import zeta.android.apps.ZetaApplication
+import zeta.android.apps.presenter.NavigationPresenter
+import zeta.android.apps.ui.activity.navigation.NavigationFragmentManager
+import zeta.android.apps.ui.common.BaseViews
+import zeta.android.apps.ui.fragment.DebugFragment
+import zeta.android.apps.ui.presentation.NavigationPresentation
+
+
+class NavigationActivity : BaseNavigationActivity(), NavigationPresentation {
+
+    private var mViews: Views? = null
+
+    @Inject lateinit var mPresenter: NavigationPresenter
+
+    internal class Views(root: AppCompatActivity) : BaseViews(root.findViewById(R.id.zeta_drawer_layout)) {
```

Comment:
:) Yea, as time permits I will try to update this project with some kotlin goodies